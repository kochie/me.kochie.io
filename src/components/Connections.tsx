import * as React from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faLinkedin,
  faTwitter,
  faInstagram,
  faGithub,
  faMediumM,
  faGitlab
} from "@fortawesome/free-brands-svg-icons";

import "../styles/Connections.scss";

const connections = [
  {
    icon: faLinkedin,
    href: "https://linkedin.com/in/rkkochie",
    name: "rkkochie",
    color: "linkedin"
  },
  {
    icon: faTwitter,
    href: "https://twitter.com/rkkochie",
    name: "rkkochie",
    color: "twitter"
  },
  {
    icon: faInstagram,
    href: "https://instagram.com/rkkochie",
    name: "rkkochie",
    color: "instagram"
  },
  {
    icon: faGithub,
    href: "https://github.com/kochie",
    name: "kochie",
    color: "github"
  },
  {
    icon: faMediumM,
    href: "https://medium.com/@kochie",
    name: "kochie",
    color: "medium"
  },
  {
    icon: faGitlab,
    href: "https://gitlab.com/kochie",
    name: "kochie",
    color: "gitlab"
  }
];

export default class Connections extends React.Component {
  render() {
    return (
      <div className="connections">
        {connections.map(connection => (
          <div className="grow">
            <a href={connection.href} className={`${connection.color} grow`}>
              <FontAwesomeIcon
                icon={connection.icon}
                //color={connection.color}
                size="2x"
                fixedWidth
                className="vertical-middle"
              />
              {/* <span>{connection.name}</span> */}
            </a>
          </div>
        ))}
      </div>
    );
  }
}
