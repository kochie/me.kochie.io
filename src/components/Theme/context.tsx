'use client'
import React, {
  PropsWithChildren,
  ReactElement,
  createContext,
  useCallback,
  useContext,
  useEffect,
  useRef,
  useState,
} from 'react'

enum THEME {
  dark = 'DARK',
  light = 'LIGHT',
  system = 'SYSTEM',
}

interface ThemeStateContext {
  setTheme: (theme: THEME) => void
  theme: THEME
}

const ThemeContext = createContext<ThemeStateContext>({
  theme: THEME.dark,
  setTheme: () => ({}),
})

const ThemeProvider = ({
  children,
}: PropsWithChildren<Record<never, never>>): ReactElement => {
  const [theme, _setTheme] = useState<THEME>(THEME.system)
  const ref = useRef(theme)

  const setTheme = (theme: THEME): void => {
    ref.current = theme
    _setTheme(theme)
  }

  const value = { theme, setTheme }

  const switchToLight = useCallback(
    (e: MediaQueryListEvent): void => {
      if (e.matches && ref.current === THEME.system) {
        document.body.classList.remove('dark-theme')
        document.body.classList.remove('dark')
      }
    },
    [ref]
  )

  const switchToDark = useCallback(
    (e: MediaQueryListEvent): void => {
      if (e.matches && ref.current === THEME.system) {
        document.body.classList.add('dark-theme')
        document.body.classList.add('dark')
      }
    },
    [ref]
  )

  useEffect(() => {
    window
      .matchMedia('(prefers-color-scheme: light)')
      .addEventListener('change', switchToLight)
    window
      .matchMedia('(prefers-color-scheme: dark)')
      .addEventListener('change', switchToDark)

    return (): void => {
      window
        .matchMedia('(prefers-color-scheme: light)')
        .removeEventListener('change', switchToLight)
      window
        .matchMedia('(prefers-color-scheme: dark)')
        .removeEventListener('change', switchToDark)
    }
  }, [switchToDark, switchToLight])

  useEffect(() => {
    const theme = window.localStorage.getItem('theme')
    if (theme) setTheme(theme as THEME)
  }, [])

  useEffect(() => {
    switch (theme) {
      case THEME.dark: {
        document.body.classList.add('dark-theme')
        document.body.classList.add('dark')
        break
      }
      case THEME.light: {
        document.body.classList.remove('dark-theme')
        document.body.classList.remove('dark')
        break
      }
      case THEME.system: {
        if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
          document.body.classList.add('dark-theme')
          document.body.classList.add('dark')
        }
        if (window.matchMedia('(prefers-color-scheme: light)').matches) {
          document.body.classList.remove('dark-theme')
          document.body.classList.remove('dark')
        }
        break
      }
    }

    window.localStorage.setItem('theme', theme)
  }, [theme])

  return <ThemeContext.Provider value={value}>{children}</ThemeContext.Provider>
}

const useTheme = (): [THEME, (theme: THEME) => void] => {
  const { theme, setTheme } = useContext(ThemeContext)
  return [theme, setTheme]
}

export { useTheme, ThemeProvider, THEME }
