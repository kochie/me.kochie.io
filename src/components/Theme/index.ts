export { default as ThemeButton } from './ThemeButton'
export { useTheme, THEME, ThemeProvider } from './context'
