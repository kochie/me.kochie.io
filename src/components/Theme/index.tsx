import React, { ReactElement, useState, useEffect, useRef } from 'react'

import styles from './theme.module.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { findIconDefinition } from '@fortawesome/fontawesome-svg-core'

enum THEME {
    dark = "DARK",
    light = "LIGHT"
}

export default function Theme(): ReactElement {
    const isDarkMode = (): boolean => {
        if (window.localStorage.getItem('theme') === null) {
            return window.matchMedia('(prefers-color-scheme: dark)').matches
        }
        return window.localStorage.getItem('theme') === THEME.dark
    }
    
    const currentIcon = useRef(null)

    const [theme, setTheme] = useState(null)
    const [animateMoon, setMoonAnimation] = useState(false)
    const [animateSun, setSunAnimation] = useState(false)
    const bulbOff = findIconDefinition({
        prefix: 'fad',
        iconName: 'lightbulb-slash',
    })

    const bulbOn = findIconDefinition({
        prefix: 'fad',
        iconName: 'lightbulb-on',
    })

    const toggleMode = (newTheme: THEME): void => {
        if (newTheme === THEME.light) {
            document.body.classList.remove('dark-theme')
        } else if (newTheme === THEME.dark) {
            document.body.classList.add('dark-theme')
        }

            // setMoonAnimation(!anima)
            // setSunAnimation(!animateSun)
        setTheme(newTheme)
    };

    useEffect(() => {
        if (isDarkMode()) document.body.classList.add('dark-theme')

        setTheme(isDarkMode() ? THEME.dark : THEME.light)

        const switchToLight = (e: MediaQueryListEvent): void => e.matches && toggleMode(THEME.light)
        const switchToDark = (e: MediaQueryListEvent): void => e.matches && toggleMode(THEME.dark)

        window.matchMedia('(prefers-color-scheme: dark)').addListener(switchToDark);
        window.matchMedia('(prefers-color-scheme: light)').addListener(switchToLight);

        return (): void => {
            window.matchMedia('(prefers-color-scheme: light)').removeListener(switchToDark);
            window.matchMedia('(prefers-color-scheme: dark)').removeListener(switchToLight);
        }
    }, [])

    useEffect(() => {
        window.localStorage.setItem('theme', theme)
    }, [theme])

    const sunDiv = (
        <div onClick={(): void => {
            setSunAnimation(true)
            toggleMode(THEME.dark)
        }} onAnimationEnd={():void => setSunAnimation(false)} className={`${styles.iconDiv}`}>
            <FontAwesomeIcon icon={bulbOff} size={'2x'} transform={{x: 5, y: 7.5}}/>
        </div>
    )
    const moonDiv = (
        <div onClick={(): void => {
            setMoonAnimation(true)
            toggleMode(THEME.light)
        }} onAnimationEnd={():void => setMoonAnimation(false)} className={`${styles.iconDiv}`}>
            <FontAwesomeIcon icon={bulbOn} size={'2x'} transform={{x: 5, y: 7.5}}/>
        </div>
    )

    return (
        <div className={styles.button} ref={currentIcon}>
            {theme === THEME.light ? sunDiv : null}
            {theme === THEME.dark ? moonDiv : null}
        </div>
    )
}