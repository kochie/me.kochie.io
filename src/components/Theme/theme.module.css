.button {
  position: fixed;
  width: 60px;
  height: 60px;
  border-radius: 50%;
  /* color: red; */
  background-color: var(--icon-color);

  text-align: center;
  /* vertical-align: middle; */
  /* line-height: 50px; */
  margin: 18px;

  display: flex;
  justify-content: center;
  align-items: center;

  top: 0;
  right: 0;

  box-shadow: rgba(0, 0, 0, 0.4) 0px 4px 8px;
  cursor: pointer;

  --fa-primary-color: var(--bulb-primary);
  --fa-secondary-color: var(--bulb-secondary);

  /* transition: 0.3s; */
  animation: opacityDown 0.4s;
}

@keyframes opacityUp {
  from {
    --fa-primary-opacity: 1;
    --fa-secondary-opacity: 0.4;
  }

  40% {
    --fa-primary-opacity: 1;
    --fa-secondary-opacity: 0.4;
  }
  50% {
    --fa-primary-opacity: 0.9;
    --fa-secondary-opacity: 0.5;
  }
  60% {
    --fa-primary-opacity: 0.8;
    --fa-secondary-opacity: 0.6;
  }
  70% {
    --fa-primary-opacity: 0.7;
    --fa-secondary-opacity: 0.7;
  }
  80% {
    --fa-primary-opacity: 0.6;
    --fa-secondary-opacity: 0.8;
  }
  90% {
    --fa-primary-opacity: 0.5;
    --fa-secondary-opacity: 0.9;
  }
  100% {
    --fa-primary-opacity: 0.4;
    --fa-secondary-opacity: 1;
  }

  to {
    --fa-primary-opacity: 0.4;
    --fa-secondary-opacity: 1;
  }
}

@keyframes opacityDown {
  from {
    --fa-primary-opacity: 0.4;
    --fa-secondary-opacity: 1;
  }

  40% {
    --fa-primary-opacity: 0.4;
    --fa-secondary-opacity: 1;
  }
  50% {
    --fa-primary-opacity: 0.5;
    --fa-secondary-opacity: 0.9;
  }
  60% {
    --fa-primary-opacity: 0.6;
    --fa-secondary-opacity: 0.8;
  }
  70% {
    --fa-primary-opacity: 0.7;
    --fa-secondary-opacity: 0.7;
  }
  80% {
    --fa-primary-opacity: 0.8;
    --fa-secondary-opacity: 0.6;
  }
  90% {
    --fa-primary-opacity: 0.9;
    --fa-secondary-opacity: 0.5;
  }
  100% {
    --fa-primary-opacity: 1;
    --fa-secondary-opacity: 0.4;
  }

  to {
    --fa-primary-opacity: 1;
    --fa-secondary-opacity: 0.4;
  }
}

.button:hover {
  animation: opacityUp 0.4s;
  --fa-primary-opacity: 0.4;
  --fa-secondary-opacity: 1;
}

.iconDiv {
  width: 100%;
  /* height: 100%; */
}
