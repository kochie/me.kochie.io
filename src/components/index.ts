export { default as App } from './App/App'
export { default as ContactButton } from './ContactButton/ContactButton'
export { default as Connections } from './Connections/Connections'
export { default as BuyMeACoffee } from './BuyMeACoffee/widget'
export { default as Button } from './Button/Button'
export { default as Footer } from './Footer/Footer'
export { default as RecentArticles } from './RecentArticles/RecentArticles'
export { default as Portfolio } from './Portfolio/Portfolio'
// export { default as Header } from "./Header/Header";
export { default as Middle } from './Middle/Middle'
export { default as Contact } from './Contact/Contact'
export { default as Sidebar } from './Sidebar/Sidebar'
export { default as Main } from './Main/Main'
export { default as Head } from './Head/Head'
export { default as Divider } from './Divider/Divider'
export { default as Theme } from './Theme'
