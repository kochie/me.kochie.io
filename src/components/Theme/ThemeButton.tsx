'use client'
import React from 'react'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  faLightbulbSlash,
  faLightbulbOn,
  faCogs,
} from '@fortawesome/pro-duotone-svg-icons'

import { THEME, useTheme } from './context'
import { faCircle } from '@fortawesome/free-solid-svg-icons'

const ThemeButton = () => {
  const [theme, setTheme] = useTheme()

  const bulbOffDiv = (
    <span
      onClick={() => setTheme(THEME.dark)}
      className="fa-stack fa-2x"
      title="Dark Theme"
      role="button"
      aria-label="Dark Theme"
      tabIndex={0}
    >
      <FontAwesomeIcon
        icon={faCircle}
        className="fa-stack-2x dark:text-white text-black"
      />
      <FontAwesomeIcon icon={faLightbulbSlash} className="fa-stack-1x" />
    </span>
  )
  const bulbOnDiv = (
    <div
      onClick={(): void => setTheme(THEME.light)}
      className="fa-stack fa-2x"
      title={'Light Theme'}
      role="button"
      aria-label="Light Theme"
      tabIndex={0}
    >
      <FontAwesomeIcon
        icon={faCircle}
        className="fa-stack-2x dark:text-white text-black"
      />
      <FontAwesomeIcon icon={faLightbulbOn} className="fa-stack-1x" />
    </div>
  )
  const systemDiv = (
    <div
      onClick={(): void => setTheme(THEME.system)}
      className="fa-stack fa-2x"
      title={'System Theme'}
      role="button"
      aria-label="System Theme"
      tabIndex={0}
    >
      <FontAwesomeIcon
        icon={faCircle}
        className="fa-stack-2x dark:text-white opacity-100 text-black"
      />
      <FontAwesomeIcon icon={faCogs} className="fa-stack-1x" />
    </div>
  )

  const nextTheme = (theme: THEME) => {
    if (theme === THEME.dark) return THEME.light
    if (theme === THEME.light) return THEME.system
    if (theme === THEME.system) return THEME.dark

    throw new Error('Undefined Theme')
  }

  return (
    <div className="fixed top-0 right-0 z-50">
      <div className="pr-5 pt-5 pl-10 pb-10 animate duration-300 group flex-col flex gap-4 items-center text-white dark:text-black">
        <div className="">
          <div
            className="w-full h-full animate duration-300"
            onClick={() => setTheme(nextTheme(theme))}
            role="button"
            aria-label="Change Theme"
            tabIndex={0}
          >
            {theme === THEME.light ? bulbOnDiv : null}
            {theme === THEME.dark ? bulbOffDiv : null}
            {theme === THEME.system ? systemDiv : null}
          </div>
        </div>

        <div className="cursor-pointer transition transform-gpu duration-300 animate scale-0 group-hover:scale-100 ease-in-out delay-200 group-hover:delay-0">
          <div
            onClick={(): void => setTheme(THEME.dark)}
            className="fa-stack duration-300 animate fa-lg"
            title={'Dark Theme'}
          >
            <FontAwesomeIcon
              icon={faCircle}
              className="fa-stack-2x dark:text-white text-black"
            />
            <FontAwesomeIcon icon={faLightbulbSlash} className="fa-stack-1x" />
          </div>
        </div>

        <div className="cursor-pointer transition transform-gpu duration-300 animate scale-0 group-hover:scale-100 ease-in-out delay-100">
          <div
            onClick={(): void => setTheme(THEME.light)}
            className="fa-stack duration-300 animate fa-lg"
            title={'Light Theme'}
          >
            <FontAwesomeIcon
              icon={faCircle}
              className="fa-stack-2x dark:text-white text-black"
            />
            <FontAwesomeIcon icon={faLightbulbOn} className="fa-stack-1x" />
          </div>
        </div>

        <div className="cursor-pointer transition transform-gpu duration-300 animate scale-0 group-hover:scale-100 ease-in-out delay-0 group-hover:delay-200">
          <div
            onClick={(): void => setTheme(THEME.system)}
            className="fa-stack duration-300 animate fa-lg"
            title={'System Theme'}
          >
            <FontAwesomeIcon
              icon={faCircle}
              className="fa-stack-2x dark:text-white text-black"
            />
            <FontAwesomeIcon icon={faCogs} className="fa-stack-1x" />
          </div>
        </div>
      </div>
    </div>
  )
}

export default ThemeButton
