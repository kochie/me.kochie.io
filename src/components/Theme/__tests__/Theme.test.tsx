import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'

import { ThemeButton, ThemeProvider } from '@/components/Theme'
import { library } from '@fortawesome/fontawesome-svg-core'
import {
  faLightbulbOn,
  faLightbulbSlash,
  faCogs,
} from '@fortawesome/pro-duotone-svg-icons'

beforeAll(() => {
  library.add(faLightbulbSlash, faLightbulbOn, faCogs)

  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // deprecated
      removeListener: jest.fn(), // deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
})

describe('THEME COMPONENT', () => {
  test('should render', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<ThemeButton />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('THEMEPROVIDER COMPONENT', () => {
  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<ThemeProvider />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
