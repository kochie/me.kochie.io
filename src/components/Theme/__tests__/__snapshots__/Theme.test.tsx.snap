// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THEME COMPONENT should render 1`] = `
<div
  className="fixed top-0 right-0 z-50"
>
  <div
    className="pr-5 pt-5 pl-10 pb-10 animate duration-300 group flex-col flex gap-4 items-center text-white dark:text-black"
  >
    <div
      className=""
    >
      <div
        aria-label="Change Theme"
        className="w-full h-full animate duration-300"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-label="Dark Theme"
          className="fa-stack fa-2x"
          onClick={[Function]}
          role="button"
          tabIndex={0}
          title="Dark Theme"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-circle fa-stack-2x dark:text-white text-black"
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
              fill="currentColor"
              style={{}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-lightbulb-slash fa-stack-1x"
            data-icon="lightbulb-slash"
            data-prefix="fad"
            focusable="false"
            role="img"
            style={{}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              className="fa-duotone-group"
              style={{}}
            >
              <path
                className="fa-secondary"
                d="M435.9 316.3c4.4-6.2 8.9-12.4 13.4-18.6l0 0 0 0 0 0c5.2-7.1 10.4-14.2 15.4-21.4c19.8-28.5 31.4-63 31.4-100.3C496 78.8 417.2 0 320 0C249.7 0 189.1 41.2 160.8 100.8l52.4 41.1C227.7 96.7 270 64 320 64c8.8 0 16 7.2 16 16s-7.2 16-16 16c-40 0-73.1 29.3-79 67.6L435.9 316.3zM152.4 229.8c5.4 16.8 13.2 32.4 23 46.6c5 7.2 10.2 14.3 15.4 21.4l0 0c19.8 27.1 39.7 54.4 49.2 86.2H348.1L152.4 229.8zM240 416v16c0 44.2 35.8 80 80 80s80-35.8 80-80v-7.1L388.7 416H240z"
                fill="currentColor"
                style={{}}
              />
              <path
                className="fa-primary"
                d="M5.1 9.2C13.3-1.2 28.4-3.1 38.8 5.1l592 464c10.4 8.2 12.3 23.3 4.1 33.7s-23.3 12.3-33.7 4.1L9.2 42.9C-1.2 34.7-3.1 19.6 5.1 9.2z"
                fill="currentColor"
                style={{}}
              />
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      className="cursor-pointer transition transform-gpu duration-300 animate scale-0 group-hover:scale-100 ease-in-out delay-200 group-hover:delay-0"
    >
      <div
        className="fa-stack duration-300 animate fa-lg"
        onClick={[Function]}
        title="Dark Theme"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-circle fa-stack-2x dark:text-white text-black"
          data-icon="circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-lightbulb-slash fa-stack-1x"
          data-icon="lightbulb-slash"
          data-prefix="fad"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            className="fa-duotone-group"
            style={{}}
          >
            <path
              className="fa-secondary"
              d="M435.9 316.3c4.4-6.2 8.9-12.4 13.4-18.6l0 0 0 0 0 0c5.2-7.1 10.4-14.2 15.4-21.4c19.8-28.5 31.4-63 31.4-100.3C496 78.8 417.2 0 320 0C249.7 0 189.1 41.2 160.8 100.8l52.4 41.1C227.7 96.7 270 64 320 64c8.8 0 16 7.2 16 16s-7.2 16-16 16c-40 0-73.1 29.3-79 67.6L435.9 316.3zM152.4 229.8c5.4 16.8 13.2 32.4 23 46.6c5 7.2 10.2 14.3 15.4 21.4l0 0c19.8 27.1 39.7 54.4 49.2 86.2H348.1L152.4 229.8zM240 416v16c0 44.2 35.8 80 80 80s80-35.8 80-80v-7.1L388.7 416H240z"
              fill="currentColor"
              style={{}}
            />
            <path
              className="fa-primary"
              d="M5.1 9.2C13.3-1.2 28.4-3.1 38.8 5.1l592 464c10.4 8.2 12.3 23.3 4.1 33.7s-23.3 12.3-33.7 4.1L9.2 42.9C-1.2 34.7-3.1 19.6 5.1 9.2z"
              fill="currentColor"
              style={{}}
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      className="cursor-pointer transition transform-gpu duration-300 animate scale-0 group-hover:scale-100 ease-in-out delay-100"
    >
      <div
        className="fa-stack duration-300 animate fa-lg"
        onClick={[Function]}
        title="Light Theme"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-circle fa-stack-2x dark:text-white text-black"
          data-icon="circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-lightbulb-on fa-stack-1x"
          data-icon="lightbulb-on"
          data-prefix="fad"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            className="fa-duotone-group"
            style={{}}
          >
            <path
              className="fa-secondary"
              d="M449.2 297.8c-19.8 27.1-39.7 54.4-49.2 86.2H240c-9.6-31.9-29.5-59.1-49.2-86.2c-5.2-7.1-10.4-14.2-15.4-21.4C155.6 247.9 144 213.3 144 176C144 78.8 222.8 0 320 0s176 78.8 176 176c0 37.3-11.6 71.9-31.4 100.3c-5 7.2-10.2 14.3-15.4 21.4zM240 176c0-44.2 35.8-80 80-80c8.8 0 16-7.2 16-16s-7.2-16-16-16c-61.9 0-112 50.1-112 112c0 8.8 7.2 16 16 16s16-7.2 16-16z"
              fill="currentColor"
              style={{}}
            />
            <path
              className="fa-primary"
              d="M69.3 4l48 32c11 7.4 14 22.3 6.7 33.3s-22.3 14-33.3 6.7l-48-32c-11-7.4-14-22.3-6.7-33.3S58.3-3.3 69.3 4zM597.3 44l-48 32c-11 7.4-25.9 4.4-33.3-6.7s-4.4-25.9 6.7-33.3l48-32c11-7.4 25.9-4.4 33.3 6.7s4.4 25.9-6.7 33.3zM24 160H88c13.3 0 24 10.7 24 24s-10.7 24-24 24H24c-13.3 0-24-10.7-24-24s10.7-24 24-24zm528 0h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H552c-13.3 0-24-10.7-24-24s10.7-24 24-24zM117.3 332l-48 32c-11 7.4-25.9 4.4-33.3-6.7s-4.4-25.9 6.7-33.3l48-32c11-7.4 25.9-4.4 33.3 6.7s4.4 25.9-6.7 33.3zm432-39.9l48 32c11 7.4 14 22.3 6.7 33.3s-22.3 14-33.3 6.7l-48-32c-11-7.4-14-22.3-6.7-33.3s22.3-14 33.3-6.7zM400 432c0 44.2-35.8 80-80 80s-80-35.8-80-80V384H400v48z"
              fill="currentColor"
              style={{}}
            />
          </g>
        </svg>
      </div>
    </div>
    <div
      className="cursor-pointer transition transform-gpu duration-300 animate scale-0 group-hover:scale-100 ease-in-out delay-0 group-hover:delay-200"
    >
      <div
        className="fa-stack duration-300 animate fa-lg"
        onClick={[Function]}
        title="System Theme"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-circle fa-stack-2x dark:text-white text-black"
          data-icon="circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512z"
            fill="currentColor"
            style={{}}
          />
        </svg>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-gears fa-stack-1x"
          data-icon="gears"
          data-prefix="fad"
          focusable="false"
          role="img"
          style={{}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            className="fa-duotone-group"
            style={{}}
          >
            <path
              className="fa-secondary"
              d="M529.7 506.7c-8.7 3.7-18.6 .9-25-6.2l-19.8-22.2c-6.8 1.1-13.8 1.7-20.9 1.7s-14.1-.6-20.9-1.7l-19.8 22.2c-6.3 7.1-16.2 9.9-25 6.2c-5.3-2.3-10.5-4.8-15.5-7.6l-5.4-3.1c-5-3-9.9-6.3-14.6-9.8c-7.6-5.7-10.1-15.7-7.1-24.7l9.3-28.2c-8.8-10.7-16-23-20.9-36.2L315.1 391c-9.3-1.9-16.7-9.1-17.8-18.5c-.8-6.7-1.2-13.5-1.2-20.4s.4-13.7 1.2-20.4c1.1-9.4 8.6-16.6 17.8-18.5l29.1-6.1c5-13.3 12.1-25.5 20.9-36.2l-9.3-28.2c-3-9-.5-19 7.1-24.7c4.7-3.5 9.6-6.8 14.6-9.9l5.2-3c5-2.8 10.2-5.3 15.6-7.6c8.7-3.7 18.6-.9 25 6.2l19.8 22.2c6.8-1.1 13.8-1.7 20.9-1.7s14.1 .6 20.9 1.7l19.8-22.2c6.3-7.1 16.2-9.9 25-6.2c5.3 2.3 10.5 4.8 15.6 7.6l5.3 3c5.1 3 9.9 6.3 14.6 9.9c7.6 5.7 10.1 15.7 7.1 24.7l-9.3 28.2c8.8 10.7 16 23 20.9 36.2l29.1 6.1c9.3 1.9 16.7 9.1 17.8 18.5c.8 6.7 1.2 13.5 1.2 20.4s-.4 13.7-1.2 20.4c-1.1 9.4-8.6 16.6-17.8 18.5L583.9 397c-5 13.3-12.1 25.5-20.9 36.2l9.3 28.2c3 9 .5 19-7.1 24.7c-4.7 3.5-9.5 6.8-14.6 9.8l-5.4 3.1c-5 2.8-10.2 5.3-15.5 7.6zM416 352a48 48 0 1 0 96 0 48 48 0 1 0 -96 0z"
              fill="currentColor"
              style={{}}
            />
            <path
              className="fa-primary"
              d="M314.7 110.3c3.7 8.7 .9 18.6-6.2 25l-22.2 19.8c1.1 6.8 1.7 13.8 1.7 20.9s-.6 14.1-1.7 20.9l22.2 19.8c7.1 6.3 9.9 16.2 6.2 25c-2.3 5.3-4.8 10.5-7.6 15.5l-3.1 5.4c-3 5-6.3 9.9-9.8 14.6c-5.7 7.6-15.7 10.1-24.7 7.1l-28.2-9.3c-10.7 8.8-23 16-36.2 20.9L199 324.9c-1.9 9.3-9.1 16.7-18.5 17.8c-6.7 .8-13.5 1.2-20.4 1.2s-13.7-.4-20.4-1.2c-9.4-1.1-16.6-8.6-18.5-17.8L115 295.9c-13.3-5-25.5-12.1-36.2-20.9l-28.2 9.3c-9 3-19 .5-24.7-7.1c-3.5-4.7-6.8-9.6-9.9-14.6l-3-5.2c-2.8-5-5.3-10.2-7.6-15.6c-3.7-8.7-.9-18.6 6.2-25l22.2-19.8C32.6 190.1 32 183.1 32 176s.6-14.1 1.7-20.9L11.5 135.3c-7.1-6.3-9.9-16.2-6.2-25c2.3-5.3 4.8-10.5 7.6-15.6l3-5.3c3-5.1 6.3-9.9 9.9-14.6c5.7-7.6 15.7-10.1 24.7-7.1l28.2 9.3c10.7-8.8 23-16 36.2-20.9L121 27.1c1.9-9.3 9.1-16.7 18.5-17.8C146.3 8.4 153.1 8 160 8s13.7 .4 20.4 1.2c9.4 1.1 16.6 8.6 18.5 17.8L205 56.1c13.3 5 25.5 12.1 36.2 20.9l28.2-9.3c9-3 19-.5 24.7 7.1c3.5 4.7 6.8 9.5 9.8 14.6l3.1 5.4c2.8 5 5.3 10.2 7.6 15.5zM160 224a48 48 0 1 0 0-96 48 48 0 1 0 0 96z"
              fill="currentColor"
              style={{}}
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`THEMEPROVIDER COMPONENT renders correctly 1`] = `null`;
